statisticsapp/src/main/java/com/example/statisticsapp/data/usecase/GetPlantsUseCaseImpl.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.statisticsapp.data.usecase
+
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantstatscontract.PlantStatisticsContract
+import com.example.statisticsapp.domain.usecase.GetPlantsUseCase
+import dagger.hilt.android.qualifiers.ApplicationContext
+import javax.inject.Inject
+
+class GetPlantsUseCaseImpl @Inject constructor(
+    @ApplicationContext private val appContext: Context
+) : GetPlantsUseCase {
+
+    override suspend fun invoke(): List<Plant> {
+        return appContext.contentResolver.query(
+            Uri.parse(PlantStatisticsContract.CONTENT_URI),
+            null,
+            null,
+            null,
+            null
+        )
+            ?.let { cursor ->
+                val plants = cursor.getPlants()
+                cursor.close()
```

Comment:
with Closable you can Use Kotlin's standart library method .use{} instead of calling .close every time